mongo/client_encryption.go

```diff
@@ -42,6 +43,40 @@ func NewClientEncryption(keyVaultClient *Client, opts ...*options.ClientEncrypti
 	db, coll := splitNamespace(ceo.KeyVaultNamespace)
 	ce.keyVaultColl = ce.keyVaultClient.Database(db).Collection(coll, keyVaultCollOpts)
 
+	credentialCallback := func() interface{} {
+		// By default, do nothing.
+		return nil
+	}
+
+	awsEmpty := false
+	if awsMap, ok := ceo.KmsProviders["aws"]; ok {
+		awsEmpty = true
+		for range awsMap {
+			awsEmpty = false
+			break
+		}
+	}
+
+	// If an empty "aws" map is configured, the driver is responsible for fetching "aws" credentials.
+	if awsEmpty {
+		// Remove the "aws" field from the map. libmongocrypt considers an empty "aws": {} an error.
+		delete(ceo.KmsProviders, "aws")
```

Comment:
I'm not clear on why it's better to remove the aws provider rather than pass through the empty map.

Reply:
See the conversation here: https://github.com/mongodb/libmongocrypt/pull/252#issuecomment-1049815015. I think it is a reasonable improvement to have libmongocrypt handle the `aws: {}` KMS provider. But it is not necessary either to meet the requirements.